packages/sanity/src/core/store/_legacy/document/document-pair/serverOperations/delete.ts

```diff
@@ -3,16 +3,33 @@ import {isLiveEditEnabled} from '../utils/isLiveEditEnabled'
 
 export const del: OperationImpl<[], 'NOTHING_TO_DELETE'> = {
   disabled: ({snapshots}) => (snapshots.draft || snapshots.published ? false : 'NOTHING_TO_DELETE'),
-  execute: ({client: globalClient, schema, idPair, typeName}) => {
+  execute: ({client: globalClient, schema, idPair, typeName, snapshots}) => {
     if (isLiveEditEnabled(schema, typeName)) {
       const tx = globalClient.observable.transaction().delete(idPair.publishedId)
       return tx.commit({tag: 'document.delete'})
     }
 
     const vXClient = globalClient.withConfig({apiVersion: 'X'})
-
     const {dataset} = globalClient.config()
 
+    //the delete action requires a published doc -- discard if not present
+    if (!snapshots.published) {
+      return vXClient.observable.request({
+        url: `/data/actions/${dataset}`,
+        method: 'post',
+        tag: 'document.delete',
```

Comment:
I've maintained the tag as delete so we have a sense of what the user's intent was. We can change it to discard if we don't like this choice.

Reply:
I don't see a problem with changing this to "document.discard", The fact that the button says "Delete" in this case feels more coincidental, and you could argue that it might as well say "Discard draft" when there is no published document.